.github/workflows/unit-tests.yml

```diff
@@ -8,29 +8,37 @@ on:
 jobs:
   unit-tests:
     runs-on: ubuntu-latest
+    strategy:
+      fail-fast: false
+      matrix:
+        python:
+          - "3.10"
```

Comment:
@nathan-weinberg I don't think we need to run these on every PR for all python versions. That's too much right now. Can we make this matrix be run when this workflow runs nightly so regressions can be caught without too much trouble?

Reply:
Sure, I can submit a followup to alter the schedule - did you have a preference as to what Python versions should be run against PRs?

I will say these all take about the same time and they run concurrently, so I'm not quite understanding what the concern is, but happy to make the followup nonetheless.

Reply:
I’m concerned about the nightly run. Based on experience, we should keep the main branch as stable as possible. Nightly tests are typically useful for tasks that take too long to run on every PR, but this step is simple and lightweight, making it worth running on every PR. Additionally, maintaining a nightly build adds overhead for maintainers, as someone must monitor failures, which can easily be overlooked. In contrast, if an issue arises in a PR, it gets immediate attention and can be resolved directly by the PR owner.

Reply:
I also think nightlies are regression vectors. I am not sure what we'd achieve by not running these tests. What are we optimizing for?

Reply:
I've opened https://github.com/meta-llama/llama-stack/pull/1644 to hopefully alleviate some of this concern from @ashwinb while keeping the unit test coverage Seb, Ihar, and I would all prefer to keep - let's continue the conversation in that PR